src/lib/rateLimiter.ts

```diff
@@ -0,0 +1,22 @@
+import RateLimit from "express-rate-limit"
+import MemcachedStore from "rate-limit-memcached"
+import { client } from "./cache"
+import { Request } from "express"
+
+// Max 100 requests in 15 minutes from a single IP
+const max = 100
+const expiration = 15 * 60
```

Comment:
Can these be config values rather than hard-coded?

The max seems surprisingly low and the window surprisingly large (especially considering that requests from the same network will appear to us as a single IP, and requesting a single artsy.net page can generate a dozen or 2 direct requests to Metaphysics). Although, since this is scoped to just 1 query at the moment, I guess it might be enough to block our specific problem.

Finally, I wonder how we'll monitor the rate at which these `429` responses get generated. Can we easily tell from logs or ELB metrics? If not, we should record a custom metric or risk not knowing when this is occurring or when we might mistakenly be limiting legitimate requests.

Reply:
Good point, made the values configurable 👍 

The papertrail logs do show status code and if we’d accidentally limit e.g. Force then we should in the traces see those HTTP requests having failed with a 429 status code.